libpod/runtime_img.go

```diff
@@ -270,26 +283,76 @@ func (k *Image) Decompose() error {
 	return nil
 }
 
-// HasImageLocal returns a bool true if the image is already pulled
-func (k *Image) HasImageLocal() bool {
+func getTags(nameInput string) (reference.NamedTagged, bool, error) {
+	inputRef, err := reference.Parse(nameInput)
+	if err != nil {
+		return nil, false, errors.Wrapf(err, "unable to obtain tag from input name")
+	}
+	tagged, isTagged := inputRef.(reference.NamedTagged)
+
+	return tagged, isTagged, nil
+
+}
+
+// GetLocalImageName returns  the name of the image if it is local.
+// It will return an empty string and error if not found.
+func (k *Image) GetLocalImageName() (string, error) {
 	_, err := k.runtime.GetImage(k.Name)
-	if err == nil {
-		return true
+	if err != nil {
+		return "", errors.Wrapf(err, "unable to obtain local image")
+	}
+	localImages, err := k.runtime.GetImages(&ImageFilterParams{})
+	if err != nil {
+		return "", errors.Wrapf(err, "unable to obtain local image")
+	}
+	_, isTagged, err := getTags(k.Name)
+	if err != nil {
+		return "", err
+	}
+	for _, image := range localImages {
+		for _, name := range image.Names {
+			imgRef, err := reference.Parse(name)
+			if err != nil {
+				continue
+			}
+			var imageName string
+			imageNameOnly := reference.Path(imgRef.(reference.Named))
+			if isTagged {
+				imageNameTag, _, err := getTags(name)
+				if err != nil {
+					continue
+				}
+				imageName = fmt.Sprintf("%s:%s", imageNameOnly, imageNameTag.Tag())
+			} else {
+				imageName = imageNameOnly
+			}
+
+			if imageName == k.Name {
+				return name, nil
+			}
+			imageSplit := strings.Split(imageName, "/")
+			baseName := imageSplit[len(imageSplit)-1]
+			if baseName == k.Name {
+				return name, nil
+			}
+		}
 	}
 	fqname, _ := k.GetFQName()
 
 	_, err = k.runtime.GetImage(fqname)
```

Comment:
Why are we getting the error here, but not checking it or returning it?

Reply:
duno, fixed